.github/workflows/deploy.yml

```diff
@@ -0,0 +1,70 @@
+# Build the caller repo's Docker image and deploy it to Elastic Beanstalk.
```

Comment:
I've gone with `deploy.yml` for the workflow name: simple and human-readable and matches the names of the caller workflows in the app.

Technically this workflow can also be used for redeploying an app and for syncing an app's Elastic Beanstalk environment definition, so perhaps `deploy.yml` might not be broad enough. Having said that I'm not sure we actually need the redeploy functionality in this script (I think you can just redeploy the current version of an app by re-running a previous deploy workflow run) and perhaps the sync-env functionality can be moved out into a separate script. So maybe `deploy.yml` is okay?

The other possible problem with `deploy.yml` as a name is it no longer matches the name of the `eb-task.yml` workflow in this same repo. Perhaps `eb-task.yml` could be renamed to just `task.yml`? Or perhaps we could use `eb-deploy.yml` instead of just `deploy.yml`?